packages/x-data-grid-premium/src/hooks/features/chartsIntegration/useGridChartsIntegration.ts

```diff
@@ -0,0 +1,266 @@
+import * as React from 'react';
+import useEnhancedEffect from '@mui/utils/useEnhancedEffect';
+import { RefObject } from '@mui/x-internals/types';
+import useOnMount from '@mui/utils/useOnMount';
+import {
+  gridRowsLoadingSelector,
+  gridFilteredSortedRowEntriesSelector,
+} from '@mui/x-data-grid-pro';
+import {
+  GridStateInitializer,
+  useGridApiMethod,
+  useGridEvent,
+  gridColumnLookupSelector,
+  runIf,
+} from '@mui/x-data-grid-pro/internals';
+
+import type { DataGridPremiumProcessedProps } from '../../../models/dataGridPremiumProps';
+
+import { GridPrivateApiPremium } from '../../../models/gridApiPremium';
+import { GridChartsIntegrationContextValue } from '../../../models/gridChartsIntegration';
+import {
+  GridChartsIntegrationApi,
+  GridChartsIntegrationPrivateApi,
+  GridChartsIntegrationState,
+} from './gridChartsIntegrationInterfaces';
+import {
+  gridChartsConfigurationPanelOpenSelector,
+  gridChartsCategoriesSelector,
+  gridChartsSeriesSelector,
+} from './gridChartsIntegrationSelectors';
+import { useGridChartsIntegrationContext } from '../../utils/useGridChartIntegration';
+
+export const chartsIntegrationStateInitializer: GridStateInitializer<
+  Pick<
+    DataGridPremiumProcessedProps,
+    'chartsIntegration' | 'chartsConfigurationPanelOpen' | 'initialState'
+  >
+> = (state, props) => {
+  if (!props.chartsIntegration) {
+    return {
+      ...state,
+      chartsIntegration: {
+        configurationPanel: {
+          open: false,
+        },
+        categories: [],
+        series: [],
+      } as GridChartsIntegrationState,
+    };
+  }
+
+  return {
+    ...state,
+    chartsIntegration: {
+      configurationPanel: {
+        open:
+          props.chartsConfigurationPanelOpen ??
+          props.initialState?.chartsIntegration?.configurationPanel?.open ??
+          false,
+      },
+      categories: props.initialState?.chartsIntegration?.categories ?? [],
```

Comment:
What are categories used for? Just for labels, or something else?
Will there be support for multiple categories?

Reply:
We had an idea of providing something like `GroupedAxis` component for this, or this could be part of the renderer as you said. We decided to go the simpler way for now, but if we need multi-level-3-or-more grouping, it might be worth to do a custom axis renderer, at least for the itegration.